sdk/storage/Azure.Storage.Blobs/tests/PartitionedDownloaderTests.cs

```diff
@@ -236,16 +256,15 @@ public MockDataSource(int length)
                 _length = length;
             }
 
-            public async Task<Response<BlobDownloadStreamingResult>> GetStreamAsync(HttpRange range, BlobRequestConditions conditions = default, bool hash = default, CancellationToken token = default)
-            {
-                await Task.Delay(25);
-                return GetStream(range, conditions, hash, token);
-            }
-
             public HttpRange FullRange => new HttpRange(0, _length);
 
-            public Response<BlobDownloadStreamingResult> GetStream(HttpRange range, BlobRequestConditions conditions, bool hash, CancellationToken token)
+            public async Task<Response<BlobDownloadStreamingResult>> GetStream(HttpRange range, BlobRequestConditions conditions, bool hash, string operation, bool async, CancellationToken token)
             {
+                if (async)
+                {
+                    await Task.Delay(25);
+                }
```

Comment:
why do we need this?

Reply:
I'm thinking it was there to prevent the issues with no awaits in an async method, as when both methods were separate, async was just calling the sync method after a delay. Should we just suppress?